src/SimplePDPService.sol

```diff
@@ -190,24 +195,38 @@ contract SimplePDPService is PDPListener, PDPRecordKeeper, Initializable, UUPSUp
     // nextProvingPeriod checks for unsubmitted proof and emits a fault if so
     function nextProvingPeriod(uint256 proofSetId, uint256 challengeEpoch, uint256 leafCount, bytes calldata) external onlyPDPVerifier {
         receiveProofSetEvent(proofSetId, OperationType.NEXT_PROVING_PERIOD, abi.encode(challengeEpoch, leafCount));
-        // Noop when proving period not yet open
+
+        // initialize state for new proofset
+        if (provingDeadlines[proofSetId] == NO_PROVING_DEADLINE) {
+            uint256 firstDeadline = block.number + getMaxProvingPeriod();
+            if (challengeEpoch < firstDeadline - challengeWindow() || challengeEpoch > firstDeadline) {
+                revert("Next challenge epoch must fall within the next challenge window");
+            }
+            provingDeadlines[proofSetId] = firstDeadline;
+            provenThisPeriod[proofSetId] = false;
+            return;
+        }
+
+        // Revert when proving period not yet open
         // Can only get here if calling nextProvingPeriod multiple times within the same proving period
         uint256 prevDeadline = provingDeadlines[proofSetId] - getMaxProvingPeriod();
         if (block.number <= prevDeadline) {
             revert("One call to nextProvingPeriod allowed per proving period");
         }
-
+    
         uint256 periodsSkipped;
         // Proving period is open 0 skipped periods
         if (block.number <= provingDeadlines[proofSetId]) {
             periodsSkipped = 0;
         } else { // Proving period has closed possibly some skipped periods
             periodsSkipped = (block.number - (provingDeadlines[proofSetId] + 1)) / getMaxProvingPeriod();
         }
-        // ensure next challenge epoch falls within the next challenge window.
-        // The next challenge window immediately precedes the next deadline
+
         uint256 nextDeadline = provingDeadlines[proofSetId] + getMaxProvingPeriod()*(periodsSkipped+1);
-        if (challengeEpoch < nextDeadline - challengeWindow() || challengeEpoch > nextDeadline) {
+        // the proofset has become empty and provingDeadline is set inactive 
+        if (challengeEpoch == NO_CHALLENGE_SCHEDULED) {
+            nextDeadline = NO_PROVING_DEADLINE;
```

Comment:
Do the calculation on line 225 above only if this branch is not hit

Reply:
Do you like 
```solidity
        uint256 nextDeadline;
        // the proofset has become empty and provingDeadline is set inactive 
        if (challengeEpoch == NO_CHALLENGE_SCHEDULED) {
            nextDeadline = NO_PROVING_DEADLINE;
        } else {
            nextDeadline = provingDeadlines[proofSetId] + getMaxProvingPeriod()*(periodsSkipped+1);
            if (challengeEpoch < nextDeadline - challengeWindow() || challengeEpoch > nextDeadline) {
                revert("Next challenge epoch must fall within the next challenge window");
            }
        }
```
better?  I could see this being a bit more clear but with the extra nested if I'm not sure.

Reply:
This is much better

Reply:
I mean the new code you have proposed here is much better.